gsonpath-compiler/src/main/java/gsonpath/generator/standard/WriteFunctions.kt

```diff
@@ -47,11 +48,12 @@ fun createWriteMethod(elementClassName: ClassName,
 }
 
 @Throws(ProcessingException::class)
-private fun writeGsonFieldWriter(codeBlock: CodeBlock.Builder,
-                                 jsonMapping: GsonObject,
-                                 currentPath: String,
-                                 serializeNulls: Boolean,
-                                 currentFieldCount: Int): Int {
+private fun writeGsonFieldWriter(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 34 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=kotlin%3AS3776)